src/Forms/GridField/GridFieldFilterHeader.php

```diff
@@ -491,4 +504,32 @@ private function getTitle(object $inst): string
 
         return ClassInfo::shortName($inst);
     }
+
+    /**
+     * Transform search context into BasicSearchContext (preserves all relevant search settings)
+     */
+    private function getBasicSearchContext(GridField $gridField, SearchContext $searchContext): BasicSearchContext
+    {
+        // Retrieve filters settings as these can be carried over as is
+        $defaultSearchFields = $searchContext->getSearchFields();
+        $defaultFilters = $searchContext->getFilters();
+        $list = $gridField->getList();
+
+        // Carry over any search form settings
+        $basicSearchContext = BasicSearchContext::create($list->dataClass());
```

Comment:
`dataClass()` is not part of the `SS_List` interface - sorry, didn't notice this until I spun this up locally to test.
```suggestion
        $basicSearchContext = BasicSearchContext::create($gridField->getModelClass());
```

Reply:
Nice, I didn't know this method was a thing 👍